packages/compass-e2e-tests/smoke-test.ts

```diff
@@ -156,163 +157,49 @@ async function run() {
-      'IS_RHEL',
-    ])
-  );
-  writeBuildInfo(infoArgs);
-  const buildInfo = await readJson(infoArgs.out);
-
-  assertCommonBuildInfo(buildInfo);
-
-  let match:
-    | { filename: string; installer: Installer; updatable: boolean }
-    | undefined = undefined;
-
-  if (context.package === 'windows_setup') {
-    assertBuildInfoIsWindows(buildInfo);
-    // TODO
-  } else if (context.package === 'windows_msi') {
-    assertBuildInfoIsWindows(buildInfo);
-    // TODO
-  } else if (context.package === 'windows_zip') {
-    assertBuildInfoIsWindows(buildInfo);
-    // TODO
-  } else if (context.package === 'osx_dmg') {
-    assertBuildInfoIsOSX(buildInfo);
-
-    const filename = buildInfo.osx_dmg_filename;
-    match = {
-      filename,
-      installer: installMacDMG,
-      // The tests need to know whether to expect the path where auto-update
-      // works automatically or if Compass will only notify the user of an
-      // update.
-      updatable: true,
-    };
-  } else if (context.package === 'osx_zip') {
-    assertBuildInfoIsOSX(buildInfo);
-    // TODO
-  } else if (context.package === 'linux_deb') {
-    assertBuildInfoIsUbuntu(buildInfo);
-    // TODO
-  } else if (context.package === 'linux_tar') {
-    assertBuildInfoIsUbuntu(buildInfo);
-    // TODO
-  } else if (context.package === 'linux_rpm') {
-    assertBuildInfoIsRHEL(buildInfo);
-    // TODO
-  } else if (context.package === 'rhel_tar') {
-    assertBuildInfoIsRHEL(buildInfo);
-    // TODO
-  }
-
-  if (match) {
-    const pkg = {
-      // we need appName because it is the name of the executable inside the
-      // package, regardless of what the package filename is named or where it
-      // gets installed
-      appName: buildInfo.productName,
-      packageFilepath: path.join(compassDir, 'dist', match.filename),
-      // TODO: releaseFilepath once we download the most recent released version too
-      installer: match.installer,
-    };
-
-    if (!context.skipDownload) {
-      assert(
-        context.bucketName !== undefined &&
-          context.bucketKeyPrefix !== undefined
-      );
-      await fs.mkdir(path.dirname(pkg.packageFilepath), { recursive: true });
-      const url = `https://${context.bucketName}.s3.amazonaws.com/${context.bucketKeyPrefix}/${match.filename}`;
-      console.log(url);
-      await downloadFile(url, pkg.packageFilepath);
-
-      // TODO: we need to also download releaseFilepath once we have that
-    }
+  const cleanups: (() => void | Promise<void>)[] = [
+    () => {
+      console.log('Cleaning up sandbox');
+      fs.rmSync(context.sandboxPath, { recursive: true });
+    },
+  ];
+  const { kind, buildInfo, filepath } = await getTestSubject(context);
+
+  try {
+    const appName = buildInfo.productName;
+    if (kind === 'osx_dmg') {
+      const { appPath, uninstall } = installMacDMG({
+        appName,
+        filepath,
+        destinationPath: context.sandboxPath,
+      });
+      cleanups.push(uninstall);
```

Comment:
We should probably uninstall before clearing out the sandbox because if it runs an uninstaller that uninstaller will be inside the sandbox.

Reply:
Good point - deleting the sandbox should be the last cleanup task 👍